eigen3/include/ignition/math/eigen3/Util.hh

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef IGNITION_MATH_EIGEN3_UTIL_HH_
+#define IGNITION_MATH_EIGEN3_UTIL_HH_
+
+#include <vector>
+#include <Eigen/Geometry>
+#include <Eigen/Eigenvalues>
+#include <ignition/math/AxisAlignedBox.hh>
+#include <ignition/math/Matrix3.hh>
+#include <ignition/math/Pose3.hh>
+#include <ignition/math/Quaternion.hh>
+#include <ignition/math/Vector3.hh>
+#include <ignition/math/eigen3/Conversions.hh>
+#include <ignition/math/OrientedBox.hh>
+
+namespace ignition
+{
+  namespace math
+  {
+    namespace eigen3
+    {
+      inline Eigen::Matrix3d computeCovarianceMatrix(
+        const std::vector<math::Vector3d> &mesh,
+        const Eigen::Vector3d &centroid)
+      {
+        Eigen::Matrix3d covariance;
+        covariance.setZero ();
+
+        // For each 3d vertex in the mesh
+        for (auto point : mesh)
+        {
+          auto pt = convert(point);
+          pt = pt - centroid;
+
+          covariance(1, 1) += pt.y() * pt.y();
+          covariance(1, 2) += pt.y() * pt.z();
+          covariance(2, 2) += pt.z() * pt.z();
+
+          pt *= pt.x ();
+          covariance(0, 0) += pt.x();
+          covariance(0, 1) += pt.y();
+          covariance(0, 2) += pt.z();
+        }
+
+        covariance(1, 0) = covariance(0, 1);
+        covariance(2, 0) = covariance(0, 2);
+        covariance(2, 1) = covariance(1, 2);
+
+        return covariance;
+      }
+
+      /// \brief Get the oriented 3d bounding box of a mesh points using PCA
+      inline ignition::math::OrientedBoxd MeshToOrientedBox(
+        const std::vector<math::Vector3d> &_mesh)
+      {
+        math::OrientedBoxd box;
+
+        // Centroid (mean) of the mesh
+        math::Vector3d mean;
+        for (auto point : _mesh)
+          mean += point;
+        mean /= _mesh.size();
+
+        Eigen::Vector3d centroid = convert(mean);
+
+        // Covariance
+        Eigen::Matrix3d covariance = computeCovarianceMatrix(_mesh, centroid);
```

Comment:
```suggestion
        Eigen::Matrix3d covariance = computeCovarianceMatrix(_mesh, centroid);
```
The comment here is redundant since the method and variable name explain what is going on here